src/test/java/com/thangle/domain/book/BookServiceTest.java

```diff
@@ -33,7 +38,135 @@ void shouldFindAll_OK() {
         assertEquals(expected.size(), actual.size());
         assertEquals(expected.get(0).getId(), actual.get(0).getId());
         assertEquals(expected.get(0).getTitle(), actual.get(0).getTitle());
+        assertEquals(expected.get(0).getAuthor(), actual.get(0).getAuthor());
+        assertEquals(expected.get(0).getDescription(), actual.get(0).getDescription());
+        assertEquals(expected.get(0).getCreatedAt(), actual.get(0).getCreatedAt());
+        assertEquals(expected.get(0).getUpdatedAt(), actual.get(0).getUpdatedAt());
+        assertEquals(expected.get(0).getImage(), actual.get(0).getImage());
+        assertEquals(expected.get(0).getUserId(), actual.get(0).getUserId());
 
         verify(bookStore).findAll();
     }
+
+    @Test
+    void shouldFindByTitleAuthorDescription_OK() throws Exception {
+        final var book = buildBook();
+        final var expected = buildBooks();
+
+        when(bookStore.find(book.getTitle())).thenReturn(expected);
+
+        final var actual = bookService.find(book.getTitle());
+
+        assertEquals(expected, actual);
+
+        verify(bookStore).find(book.getTitle());
+    }
+
+    @Test
+    void shouldCreate_OK() {
+        final var book = buildBook();
+        when(bookStore.save(book)).thenReturn(book);
+
+        final var updatedBook = bookService.create(book);
+
+        assertEquals(book, updatedBook);
+
+        verify(bookStore).save(book);
+    }
+
+    @Test
+    void shouldCreate_WithTitleEmpty() {
+        final var book = buildBook();
+        book.setTitle("");
```

Comment:
```suggestion
        final var book = buildBook().withTitle("");
```

Và những chỗ khác tương tự